### PR TITLE
[SPARK-32946][R][SQL] Add withColumn to SparkR

### PR DESCRIPTION
**Pull request by zero323**
### What changes were proposed in this pull request?

This PR adds `withColumn` function SparkR.

### Does this PR introduce _any_ user-facing change?

Yes, new function, equivalent to Scala and PySpark equivalents, is exposed to the end user.

### How was this patch tested?

New unit tests added.

### CODE REVIEW

**Comment by zero323** [comment]
Jenkins, retest this please.

File: R/pkg/R/column.R [hunk 1]
```diff
@@ -356,3 +356,34 @@ setMethod("%<=>%",
 #' }
 #' @note ! since 2.3.0
 setMethod("!", signature(x = "Column"), function(x) not(x))
+
+#' withField
+#'
+#' Adds/replaces field in a struct `Column` by name.
```
Comment (HyukjinKwon) [inline R/pkg/R/column.R:362]: nit maybe `Column` -> either `\code{Column}` or just `column`

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

**Comment by HyukjinKwon** [comment]
Merged to master.

Thank you for taking a look for this one @zero323.

**Comment by zero323** [comment]
Thanks @HyukjinKwon

**Pull request closed without merging**